blocks/library/latest-posts/index.js

```diff
@@ -72,22 +70,16 @@ registerBlockType( 'core/latest-posts', {
 		}
 
 		componentWillReceiveProps( nextProps ) {
-			const { postsToShow: postToShowCurrent } = this.props.attributes;
-			const { postsToShow: postToShowNext } = nextProps.attributes;
-			const { setAttributes } = this.props;
-
-			if ( postToShowCurrent === postToShowNext ) {
+			if ( this.props.attributes === nextProps.attributes ) {
```

Comment:
I don't think we want to rely on a strict equality check here; these two objects may in fact be different references while containing the same values for `postsToShow`, `order`, `orderBy`, `categories`. If we need to, we can use [Lodash's `_.isEqual`](https://lodash.com/docs/4.17.4#isEqual) for a deep equality comparison.